notebooks/4_visualize_forecast.ipynb

```diff
@@ -0,0 +1,1082 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots(1, 1, figsize=(20, 20))\n",
+    "divider = make_axes_locatable(ax)\n",
+    "cax = divider.append_axes(\"right\", size=\"2%\", pad=0.1)\n",
+    "ax.set_title(\n",
+    "    \"Mean Year of First Introduction (mean of runs with intros >50%)\\n\" + run_name,\n",
+    "    fontsize=18,\n",
+    ")\n",
+    "countries_geo[\"arr_yr_mean\"] = np.where(\n",
+    "    countries_geo[\"intro_proportion\"] < 0.5, np.nan, countries_geo[\"arr_yr_mean\"]\n",
+    ")\n",
+    "countries_geo.plot(\n",
+    "    column=\"arr_yr_mean\",\n",
+    "    categorical=True,\n",
+    "    cmap=\"viridis\",\n",
+    "    legend=True,\n",
+    "    ax=ax,\n",
+    "    missing_kwds={\"color\": \"lightgrey\"},\n",
+    "    legend_kwds={\"loc\": \"lower left\"},\n",
+    ")\n",
+    "\n",
+    "plt.savefig(\n",
+    "    results_dir + \"/figs/first_intro_mean_more50pct.png\",\n",
+    "    bbox_inches=\"tight\",\n",
+    "    pad_inches=0.01,\n",
+    ")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Additional plots: histograms and heatmap of introductions by country source"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Assess bridgehead populations with a temporal heatmap:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "destinations_all = list(set(first_intros_all[\"Destination\"]))\n",
+    "min_intro_prop = 0.5\n",
+    "destinations = []\n",
+    "for i in range(len(destinations_all)):\n",
+    "    if (\n",
+    "        len(\n",
+    "            first_intros_all.loc[first_intros_all[\"Destination\"] == destinations_all[i]]\n",
+    "        )\n",
+    "        > num_runs * min_intro_prop\n",
+    "    ):\n",
+    "        destinations.append(destinations_all[i])\n",
+    "num_destinations = len(destinations)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Count origins for each timestep\n",
+    "origin_countries_by_ts = (\n",
+    "    pd.DataFrame(index=countries.iloc[:, 0], columns=sim_years)\n",
+    ").fillna(0)\n",
+    "origins = (org_dest_all.groupby([\"Origin\", \"TS\"]).count()).reset_index().fillna(0)\n",
+    "bridgehead_years = range(origins.TS.min(), origins.TS.max()+1)\n",
+    "\n",
+    "for i in range(len(origins)):\n",
+    "    origin = origins.iloc[i, :]\n",
+    "    origin_countries_by_ts.loc[origin.Origin, origin.TS] = origin.Destination"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create heatmap of bridgehead introductions\n",
+    "origin_countries_by_ts_filtered = origin_countries_by_ts[\n",
+    "    origin_countries_by_ts.sum(axis=1) > 0][bridgehead_years]\n",
```

Comment:
Add .fillna(0) to change NAN to 0s in matrix.

origin_countries_by_ts_filtered = origin_countries_by_ts[
    origin_countries_by_ts.sum(axis=1) > 0][bridgehead_years].fillna(0)